src/FSLibrary/MaxTPSTest.fs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+module MaxTPSTest
+
+// This module provides a configurable max TPS test to use in missions
+
+open Logging
+open StellarCoreHTTP
+open StellarCoreSet
+open StellarFormation
+open StellarMissionContext
+open StellarNetworkData
+open StellarStatefulSets
+open StellarSupercluster
+
+let maxTPSTest
+    (context: MissionContext)
+    (baseLoadGen: LoadGen)
+    (setupCfg: LoadGen option)
+    (increaseSorobanLimits: bool)
+    =
+    let allNodes =
+        if context.pubnetData.IsSome then
+            FullPubnetCoreSets context true false
+        else
+            StableApproximateTier1CoreSets
+                context.image
+                (if context.flatQuorum.IsSome then context.flatQuorum.Value else false)
+
+    let sdf =
+        List.find (fun (cs: CoreSet) -> cs.name.StringName = "stellar" || cs.name.StringName = "sdf") allNodes
+
+    let tier1 = List.filter (fun (cs: CoreSet) -> cs.options.tier1 = Some true) allNodes
+
+    context.ExecuteWithOptionalConsistencyCheck
+        allNodes
+        None
+        false
+        (fun (formation: StellarFormation) ->
+
+            let numAccounts = 30000
+
+            let upgradeMaxTxSetSize (coreSets: CoreSet list) (rate: int) =
+                // Set max tx size to 10x the rate -- at 5x we overflow the transaction queue too often.
+                formation.UpgradeMaxTxSetSize coreSets (10 * rate)
+
+            // Setup overlay connections first before manually closing
+            // ledger, which kick off consensus
+            formation.WaitUntilConnected allNodes
+            formation.ManualClose allNodes
+
+            // Wait until the whole network is synced before proceeding,
+            // to fail asap in case of a misconfiguration
+            formation.WaitUntilSynced allNodes
+            formation.UpgradeProtocolToLatest allNodes
+            upgradeMaxTxSetSize allNodes 10000
+            formation.RunLoadgen sdf { context.GenerateAccountCreationLoad with accounts = numAccounts }
+
+            // Upgrade limits (if requested)
+            if increaseSorobanLimits then
+                formation.UpgradeSorobanLedgerLimitsWithMultiplier allNodes 100000
```

Comment:
I'd like to open a discussion about the ledger limits here. The ledger limits have a huge impact on the max TPS because at some point the transaction rate results in hitting ledger limits and loadgen fails due to too many failed transactions. I can think of two ways of thinking about this test:

1. We want to know the max TPS under a certain real-world set of ledger limits, and therefore we want loadgen to hit these limits in finding the max TPS. In this case the ledger limit should probably be set more precisely than I'm currently setting it.
2. We want to know what the maximum load stellar-core can handle is, independently of network limits and therefore we should set the network limits to be very high so loadgen fails somewhere before hitting network limits. In this case I would need to really crank up the ledger limits here.

Reply:
@marta-lokhova and I chatted about this a bit and decided that the first step is to try setting the limits dynamically per run to be about double what a "worst case" sample from the distributions would be. If that doesn't work, then I'll dig into the overlay code to better tailor these limits.

Reply:
I made the network limits dynamic in ea939e8185a1dc3070e32a6771a786f8bac7ab94. I also set the required soroban success rate to `0%` and simulated apply time to match the distributions in `MissionSimulatePubnet`. The issue with the inability to preflight transactions was just too complex. However, the dynamic limits work well enough to ensure that the transactions all make it into the queue, even if they don't necessarily succeed at apply time.

Reply:
Also, I tested this with a small network and lower timeouts. The next step is to make sure it works on a full-sized run.